podpac/core/data/datasource.py

```diff
@@ -372,7 +372,11 @@ def _eval(self, coordinates, output=None, _selector=None):
         data = rsd
         if output is None:
             if requested_coordinates.crs.lower() != coordinates.crs.lower():
-                data = self.create_output_array(rsc, data=data.data)
+                if rsc.shape == data.shape:
+                    data = self.create_output_array(rsc, data=data.data)
+                else:
+                    crds = Coordinates.from_xarray(data, crs=data.attrs.get("crs", None))
+                    data = self.create_output_array(crds.transform(rsc.crs), data=data.data)
```

Comment:
I think this piece is needed in general... since we've relaxed the requirement that get_data returns exactly what was asked...